tests/infrastructure/golden_images/test_common_templates_data_volumes.py

```diff
@@ -39,11 +39,10 @@ def vm_from_golden_image(
     request,
     unprivileged_client,
     namespace,
-    ocs_storage_class,
     golden_image_data_source_scope_function,
 ):
-    use_ocs_storage_class = request.param.get("ocs_storage_class")
-    storage_class = ocs_storage_class.name if use_ocs_storage_class else None
+    set_storage_class = request.param.get("set_storage_class")
+    storage_class = py_config[STORAGE_CLASS_B] if set_storage_class else None
```

Comment:
for envs like gcp or oci, using `storage_class_for_storage_migration_a` and `storage_class_for_storage_migration_b` will use the same storage class.

Reply:
I agree that the global_config file have 1 storage class configured to them both which isn't good.
Having taken that into consideration, I feel like we should rally on the global_config file and not on something hard coded in the test.
Changing global_config for certain clusters architects isn't what this PR is about.

Reply:
Instead of hardcoding storage_class_for_storage_migration_*, we can dynamically query the cluster for available storage classes, check if any of them match the entries in storage_class_matrix, and select one from there.
Since storage_class_matrix already defines the supported storage classes for a given architecture, this ensures we only pick a class that's both available in the cluster and valid for the current environment.

Reply:
>I don't see the difference between setting a constant value in storage_class_matrix or storage_class_for_storage_migration_

see gcp for example - the matrix has 2 SCs and the 2 tests will run, while sc_a == sc_b

Reply:
You are right about GCP.
I will ping @jpeimer to understand why that is. to me it looks like a mistake.

Reply:
After talking with @jpeimer about the storage classes, it looks like even though we set 2 storage classes on the `storage_class_matrix`, one of the storage classes is not good enough and still need farther tests/work on.
so GCP is a good example of why we should use the  `STORAGE_CLASS_*` and not `storage_class_matrix`